src/Extensions/UserDefinedFormControllerExtension.php

```diff
@@ -53,44 +55,84 @@ public function updateEmail(Email $email, EmailRecipient $recipient): void
                 ->filter('EmailAddress', $recipient->EmailAddress)->first();
 
             if ($encryptionCertificateEntry && $encryptionCertificateEntry->EncryptionCrt->exists()) {
-                $pathToFile = $this->getFilePath($encryptionCertificateEntry->EncryptionCrt);
+                $encryptionPathFile = $this->getFilePath($encryptionCertificateEntry->EncryptionCrt);
             }
 
             // If no encryption certificate was found then proceed but append a warning to the email.
-            if (!$pathToFile) {
+            if (!$encryptionPathFile) {
                 $subject = $email->getSubject();
                 $encryptionMessage = '[UNENCRYPTED: CHECK CMS CONFIGURATION]';
 
                 $email->setSubject(sprintf('%s %s', $subject, $encryptionMessage));
             }
 
+            // Get the Sender Email address
             $senderEmailAddress = array_key_first($email->getFrom());
 
+            // Get the credentials for signing an Email
             $signingCredentials = $this->checkForSigningCredentials($senderEmailAddress);
         }
 
-        // Re-registering SmimeMailer for each recipient with appropriate encryption/signing details
-        $this->registerSMIMEMailer(
-            $pathToFile,
-            $signingCredentials
-        );
+        // Sign the email using the SMIMESigner
+        $this->signFormEmail($signingCredentials, $email);
+
+        // Encrypt the email using the SMIMEEncrypter
+        if (!$skipEncryption) {
```

Comment:
Likewise here, prior to sending the email with `Mailer::send()` we perform the _encryption_ of the email at this point.